pom.xml

```diff
@@ -2649,6 +2649,13 @@
     </profile>
 
     <profile>
+      <id>kubernetes</id>
+      <modules>
+        <module>resource-managers/kubernetes/core</module>
```

Comment:
We expected it would be a multi-module (https://github.com/apache-spark-on-k8s/spark/tree/branch-2.2-kubernetes/resource-managers/kubernetes). The other modules being - configuration files for the docker images and integration tests. The docker files are pretty much static configuration files, so, moving that instead to a directory like `conf/kubernetes/` might make sense also. I'm not sure where the integration tests will be eventually. 

cc @mccheah @ssuchter

Reply:
Based on a discussion in last week's meeting with Shane Knapp from RISELab, we want to keep the integration tests as a sub-module here - in the interest of keeping test code together. We should have the additional parent pom to facilitate that.

Reply:
@vanzin this isn't a multi-module project in the sense that the Kubernetes cluster manager and spark-submit implementation are split across multiple projects - but rather that there is a module for said cluster manager + spark-submit implementation, and then there are modules for integration testing said code.

@foxish The Dockerfiles feel more like application code rather than static configuration but that might just be a matter of implementation. The structure of the `CMD` in the Dockerfiles is particular to what `spark-submit` will expect for example.

Reply:
Why are integration tests in a separate module? e.g. maven has an `integration-test` phase which is separate from the usual `test` phase used for unit tests. And that's assuming that you really don't want to run them during unit tests. Then all code could potentially live in the same module.

> in the interest of keeping test code together

That would mean keeping the test code in the same module as the core code, not in a separate module.

Reply:
It's not absolutely necessary to have integration tests in a specific separate module. However, there are some nice organizational benefits we can get. For example, integration tests in the same module as the core code will need a specific package namespace that is omitted from the `test` phase and only executed in the `integrationTest` phase. Having a separate module means that the integration test pom can just make the `test` phase a no-op and integrationTest runs all tests in the `test` folder. (I don't know if Maven has a concept of a difference between `src/test/scala` and `src/integrationTest/scala`, which would help a lot.)

It's also IMO easier to read the `pom.xml` of the integration test separately from the `pom.xml` of the Kubernetes core implementation. FWIW this is what we have in the integration test POM at the moment: https://github.com/apache-spark-on-k8s/spark/blob/branch-2.2-kubernetes/resource-managers/kubernetes/integration-tests/pom.xml. (The minikube related things are going away with https://github.com/apache-spark-on-k8s/spark/pull/521).

> And that's assuming that you really don't want to run them during unit tests.

We definitely don't want to run these during unit tests - they are relatively expensive, require building Docker images, and require Minikube to be pre-installed on the given machine. Having them in at least the separate integration test phase makes these differences clear.

Reply:
That (keeping them separate) is actually pretty useful for SBT.